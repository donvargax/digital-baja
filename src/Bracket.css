.bracket-canvas {
    height: 100%;
    min-height: 500px;
    background-color: #3c3c3c !important;
    background-image: linear-gradient(0deg, transparent 24%, rgba(255, 255, 255, 0.05) 25%, rgba(255, 255, 255, 0.05) 26%, transparent 27%, transparent 74%, rgba(255, 255, 255, 0.05) 75%, rgba(255, 255, 255, 0.05) 76%, transparent 77%, transparent), linear-gradient(90deg, transparent 24%, rgba(255, 255, 255, 0.05) 25%, rgba(255, 255, 255, 0.05) 26%, transparent 27%, transparent 74%, rgba(255, 255, 255, 0.05) 75%, rgba(255, 255, 255, 0.05) 76%, transparent 77%, transparent);
    background-size: 50px 50px;
}

.bracket-node {
    background-color: #1e1e1e;
    border-radius: 5px;
    font-family: sans-serif;
    color: white;
    border: solid 2px black;
    overflow: visible;
    font-size: 11px;
}

.bracket-node__title {
    background: rgba(0,0,0,0.3);
    display: flex;
    white-space: nowrap;
}

.bracket-node__name {
    flex-grow: 1;
    padding: 5px 5px;
}

.bracket-node .srd-port {
    width: 0;
    height: 2px;
}