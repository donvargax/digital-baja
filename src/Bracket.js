import * as _ from "lodash";
import * as React from "react";
import { AbstractNodeFactory, AbstractPortFactory, BaseWidget, DefaultLinkModel, DefaultNodeModel, DiagramEngine, DiagramModel, DiagramWidget, NodeModel, PortModel, PortWidget } from "storm-react-diagrams";
import "storm-react-diagrams/dist/style.min.css";

import { BracketNodeWidget } from "./BracketNodeWidget"
import { BracketNodeModel } from "./BracketNodeModel"
import { BracketPortModel } from "./BracketPortModel"

import './Bracket.css'
import { Exception } from "handlebars";

// class DiamondPortModel extends PortModel {
// 	constructor(pos = "top") {
// 		super(pos, "diamond")
// 		this.position = pos
// 	}
// 	serialize() {
// 		return _.merge(super.serialize(), {
// 			position: this.position
// 		})
// 	}
// 	deSerialize(data, engine) {
// 		super.deSerialize(data, engine)
// 		this.position = data.position
// 	}
// 	createLinkModel() {
// 		return new DefaultLinkModel()
// 	}
// }

// class DiamondNodeModel extends NodeModel {
// 	constructor() {
// 		super("diamond")
// 		this.addPort(new DiamondPortModel("top"))
// 		this.addPort(new DiamondPortModel("left"))
// 		this.addPort(new DiamondPortModel("bottom"))
// 		this.addPort(new DiamondPortModel("right"))
// 	}
// }

// class DiamondNodeFactory extends AbstractNodeFactory {
// 	constructor() {
// 		super("diamond")
// 	}
// 	generateReactWidget(diagramEngine, node) {
// 		return <DiamondNodeWidget node={node} />
// 	}
// 	getNewInstance() {
// 		return new DiamondNodeModel()
// 	}
// }

class SimplePortFactory extends AbstractPortFactory {
	constructor(type, cb) {
		super(type)
		this.cb = cb
	}

	getNewInstance(initialConfig) {
		return this.cb(initialConfig)
	}
}

class BracketNodeFactory extends AbstractNodeFactory {
	constructor() {
		super("bracket")
	}
	generateReactWidget(diagramEngine, node) {
		return <BracketNodeWidget node={node} />
	}
	getNewInstance() {
		return new BracketNodeModel()
	}
}

export const Bracket = ({games = []}) => {
	const startx = 10;
	const starty = 10;
	const width = 300;
	const height = 100;
	let engine = new DiagramEngine();
	engine.installDefaultFactories();

	engine.registerPortFactory(new SimplePortFactory("bracket", config => new BracketPortModel()))
	engine.registerNodeFactory(new BracketNodeFactory())

	let model = new DiagramModel();

	let nodes = [];
	for (let i = 0; i < 4; i++) {
		let game = { };
		if (i < games.length) {
			game = games[i];
		}

		let node = new BracketNodeModel(game.home, game.visitor, "#eee", game.played, game.homeWinner)
		node.setPosition(startx, starty + height * i)
		nodes.push(node)
	}

	for (let i = 4; i < 6; i++) {
		let game = { };
		if (i < games.length) {
			game = games[i];
		}

		let node = new BracketNodeModel(game.home, game.visitor, "#eee", game.played, game.homeWinner)
		node.setPosition(startx + width, starty + (height / 2) * (((i - 3) * (i - 3)) + i - 4))
		nodes.push(node)
	}

	let game = { };
	if (7 == games.length) {
		game = games[6];
	}

	let node = new BracketNodeModel(game.home, game.visitor, "#eee", game.played, game.homeWinner)

	node.setPosition(startx + (width * 2), starty + height + height / 2)
	nodes.push(node)

	let links = []

	links.push(nodes[0].getRightPort().link(nodes[4].getLeftPort()))
	links.push(nodes[1].getRightPort().link(nodes[4].getLeftPort()))

	links.push(nodes[2].getRightPort().link(nodes[5].getLeftPort()))
	links.push(nodes[3].getRightPort().link(nodes[5].getLeftPort()))

	links.push(nodes[4].getRightPort().link(nodes[6].getLeftPort()))
	links.push(nodes[5].getRightPort().link(nodes[6].getLeftPort()))

	//4) add the models to the root graph
	model.addAll(...nodes, ...links);
	// model.addAll(node5, node6, node7)

	//5) load model into engine
	engine.setDiagramModel(model);

	//6) render the diagram!
	return <DiagramWidget className="bracket-canvas" diagramEngine={engine} />;
};