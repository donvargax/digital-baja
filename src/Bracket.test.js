import React from 'react';
import { render } from 'react-testing-library'

import { Toolkit } from "storm-react-diagrams"

import { Bracket } from './Bracket';

Toolkit.TESTING = true;
window.focus = function () { };

it('renders without crashing', () => {
    const { asFragment } = render(<Bracket />)
    expect(asFragment()).toMatchSnapshot()
});

it('renders with 1 game', () => {
    const games = [
        {
            home: "Graham, Kunde and Abshire",
            visitor: "Baumbach - Keebler",
            played: true,
            homeWinner: true,
        },
    ]
    const { asFragment } = render(<Bracket games={games} />)
    expect(asFragment()).toMatchSnapshot()
});

it('renders with multiple games', () => {
    const games = [
        {
            home: "Graham, Kunde and Abshire",
            visitor: "Baumbach - Keebler",
            played: true,
            homeWinner: true,
        },
        {
            home: "McCullough - Russel",
            visitor: "Pagac, O'Hara and Goodwin",
            played: true,
            homeWinner: false,
        },
        {
            home: "Spencer - Stanton",
            visitor: "Rodriguez - Doyle",
            played: true,
            homeWinner: true,
        },
        {
            home: "O'Connel, Schultz and Halvorson",
            visitor: "King, Miller and Haley",
            played: true,
            homeWinner: true,
        },
        {
            home: "Graham, Kunde and Abshire",
            visitor: "Pagac, O'Hara and Goodwin",
            played: true,
            homeWinner: false,
        },
        {
            home: "Spencer - Stanton",
            visitor: "O'Connel, Schultz and Halvorson",
            played: true,
            homeWinner: true,
        },
        {
            home: "Pagac, O'Hara and Goodwin",
            visitor: "Spencer - Stanton",
            played: true,
            homeWinner: false,
        },
    ]

    const { asFragment } = render(<Bracket games={games} />)
    expect(asFragment()).toMatchSnapshot()
});
