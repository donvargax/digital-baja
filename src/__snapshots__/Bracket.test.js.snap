// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with 1 game 1`] = `
<DocumentFragment>
  <div
    class="srd-diagram bracket-canvas "
  >
    <svg
      class="srd-link-layer "
      style="transform: translate(0px,0px) scale(1);"
    >
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="29"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="34"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="39"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="44"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="49"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="54"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
    </svg>
    <div
      class="srd-node-layer "
      style="transform: translate(0px,0px) scale(1);"
    >
      <div
        class="node srd-node "
        data-nodeid="8"
        style="top: 10px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
              style="background: rgb(0, 170, 0);"
            >
              Graham, Kunde and Abshire
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="8"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="8"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              Baumbach - Keebler
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="11"
        style="top: 110px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="11"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="11"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="14"
        style="top: 210px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="14"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="14"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="17"
        style="top: 310px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="17"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="17"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="20"
        style="top: 60px; left: 310px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="20"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="20"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="23"
        style="top: 260px; left: 310px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="23"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="23"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="26"
        style="top: 160px; left: 610px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="26"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="26"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders with multiple games 1`] = `
<DocumentFragment>
  <div
    class="srd-diagram bracket-canvas "
  >
    <svg
      class="srd-link-layer "
      style="transform: translate(0px,0px) scale(1);"
    >
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="29"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="34"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="39"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="44"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="49"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="54"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
    </svg>
    <div
      class="srd-node-layer "
      style="transform: translate(0px,0px) scale(1);"
    >
      <div
        class="node srd-node "
        data-nodeid="8"
        style="top: 10px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
              style="background: rgb(0, 170, 0);"
            >
              Graham, Kunde and Abshire
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="8"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="8"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              Baumbach - Keebler
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="11"
        style="top: 110px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              McCullough - Russel
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="11"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="11"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
              style="background: rgb(0, 170, 0);"
            >
              Pagac, O'Hara and Goodwin
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="14"
        style="top: 210px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
              style="background: rgb(0, 170, 0);"
            >
              Spencer - Stanton
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="14"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="14"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              Rodriguez - Doyle
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="17"
        style="top: 310px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
              style="background: rgb(0, 170, 0);"
            >
              O'Connel, Schultz and Halvorson
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="17"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="17"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              King, Miller and Haley
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="20"
        style="top: 60px; left: 310px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              Graham, Kunde and Abshire
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="20"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="20"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
              style="background: rgb(0, 170, 0);"
            >
              Pagac, O'Hara and Goodwin
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="23"
        style="top: 260px; left: 310px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
              style="background: rgb(0, 170, 0);"
            >
              Spencer - Stanton
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="23"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="23"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              O'Connel, Schultz and Halvorson
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="26"
        style="top: 160px; left: 610px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              Pagac, O'Hara and Goodwin
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="26"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="26"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
              style="background: rgb(0, 170, 0);"
            >
              Spencer - Stanton
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="srd-diagram bracket-canvas "
  >
    <svg
      class="srd-link-layer "
      style="transform: translate(0px,0px) scale(1);"
    >
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="29"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="34"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="39"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="44"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="49"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
      <g
        class="srd-default-link "
      >
        <g>
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            stroke="rgba(255,255,255,0.5)"
            stroke-width="3"
          />
          <path
            class=""
            d="M0,0 C 0,50
    0,-50 0,0"
            data-linkid="54"
            stroke="rgba(255,255,255,0.5)"
            stroke-linecap="round"
            stroke-opacity="0"
            stroke-width="20"
          />
        </g>
      </g>
    </svg>
    <div
      class="srd-node-layer "
      style="transform: translate(0px,0px) scale(1);"
    >
      <div
        class="node srd-node "
        data-nodeid="8"
        style="top: 10px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="8"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="8"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="11"
        style="top: 110px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="11"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="11"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="14"
        style="top: 210px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="14"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="14"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="17"
        style="top: 310px; left: 10px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="17"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="17"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="20"
        style="top: 60px; left: 310px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="20"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="20"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="23"
        style="top: 260px; left: 310px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="23"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="23"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
      <div
        class="node srd-node "
        data-nodeid="26"
        style="top: 160px; left: 610px;"
      >
        <div
          class="bracket-node "
          style="background: rgb(238, 238, 238);"
        >
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
          <div
            class="port srd-port "
            data-name="left"
            data-nodeid="26"
          />
          <div
            class="port srd-port "
            data-name="right"
            data-nodeid="26"
          />
          <div
            class="bracket-node__title "
          >
            <div
              class="bracket-node__name "
            >
              None
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
